tests/test_driver.sh

```diff
@@ -0,0 +1,130 @@
+# vim: set ts=4 sw=4 tw=100 et:
+# POSIX sh test driver to reduce dependency on fish in tests
+
+# macOS has really weird default IFS behavior that splits output in random places, and the trailing
+# backspace is to prevent \n from being gobbled up by the subshell output substitution.
+# Folks, this is why you should use fish!
+IFS="$(printf "\n\b")"
+
+# The first argument is the path to the script to launch; all remaining arguments are forwarded to
+# the script.
+fish_script="$1"
+shift 1
+script_args="${@}"
+
+die() {
+    if test "$#" -ge 0; then
+        printf "%s\n" "$@" 1>&2
+    fi
+    exit 1
+}
+
+# To keep things sane and to make error messages comprehensible, do not use relative paths anywhere
+# in this script. Instead, make all paths relative to one of these or $homedir."
+TESTS_ROOT="$(dirname "$0")"
+BUILD_ROOT="${TESTS_ROOT}/.."
+FISH_ROOT="${BUILD_ROOT}/.."
+
+# macOS (still) doesn't have `readlink -f` or `realpath`. It's OK, this is just for aesthetics.
+if which realpath 1>/dev/null 2>/dev/null; then
```

Comment:
`command -v`, please. `which` is unnecessary cruft and may not be installed itself.

Reply:
Sure.